packages/main-menu-react/src/MainMenu.tsx

```diff
@@ -0,0 +1,68 @@
+import React from "react";
+import { useScreen } from "@fremtind/jkl-react-hooks";
+import { Hamburger } from "@fremtind/jkl-hamburger-react";
+
+import {
+    mainMenuContext,
+    NavigationFunction,
+    IsActiveFunction,
+    defaultNavigationFunction,
+    defaultIsActiveFunction,
+} from "./contexts/mainMenuContext";
+import { MenuItemList, RootItem } from "./types";
+import { FSMenuContextProvider } from "./contexts/fullScreenMenuContext";
+import { FullScreenMenu } from "./FullScreenMenu";
+import { MainMenuLink } from "./MainMenuLink";
+
+interface MainMenuProps {
+    className?: string;
+    items: MenuItemList;
+    ariaLabel?: string;
+    showTopLevel?: boolean;
+    navigationFunction?: NavigationFunction;
+    isActiveFunction?: IsActiveFunction;
+    reduceMotion?: boolean;
+}
+
+export const MainMenu: React.FC<MainMenuProps> = ({
+    className = "",
+    items,
+    ariaLabel = "Hovedmeny",
+    showTopLevel,
+    navigationFunction = defaultNavigationFunction,
+    isActiveFunction = defaultIsActiveFunction,
+    reduceMotion = false,
+    children,
+}) => {
+    const screen = useScreen();
```

Comment:
Hva tenker vi om sånne luringer som dette. Burde det være props inn, så vi kan holde oss til bare en eventListner av samme slag, eller er det en forbedring av apiet som gjør at det er verdt å ha noen ekstra eventlistners her og der

Reply:
( patternet her er også suboptimalt for SSR, så noe vi bør se på i Fremtinden™)

Reply:
Tenker at det er verdt å ha med i starten, men er helt klart noe vi bør prøve å holde styr på. Akkurat her er det litt kjipt at man ikke kan kalle hooks conditionally. Hadde vært fint å bare sette den dersom man ikke fikk inn en prop f.eks..

Reply:
```
const screen = useEffect(() =>{
  if(hasScreen){
    return hasSreen;
  }
  return useScreen();
},[hasScreen]);
```

kan man gjøre det?

Reply:
Kunne det vært en ide at `useScreen()` tok et parameter som kan slå av event listeneren? Tanken er da at det ikke blir opprettet en en event listener før et parameter (som er default true) er satt truthy.

Det vil kanskje se litt snodig ut, men jeg tenker det gir mening ut i fra et perspektiv om å begrense antall aktive event listenere.

Men, hvor mange listeners er egentlig et problem?

Reply:
Hvor stor er en fisk?

Menja, det er ikke et problem her. Kan bli et problem i en komponent som det blir mange av. Men er jo bare å ta den inn som en required prop, så er det opp til klient å håntere det, enten med HOC, context eller bare en variabel eller konstant

Reply:
Hmm.. Dette kan vel løses slik:

```tsx
import { useScreen } from "@fremtind/jkl-react-hooks";

interface Props {
    screenInfo: typeof useScreen;
    // ...
}

export const MainMenu: React.FC<Props> = ({ screenInfo = useScreen }) => {
    const screen = screenInfo();
    // resten av greiene
}
```

Reply:
Njæ... tror ikke det vil virke, screenInfo vil ikke bli behandlet som en hook. Og om det viker, så løser det ikke problemet. Med mindre man pakker det inn på en ganske funky måte, og da vil ikke typen stemme.

Hvis vi skal støtte den oppførselen så må useScreen kunne ta inn shapen på den den normalt returner, og returnere den objectet isteden for å sette opp eventListneren.

så i useScreen
```
const useScreen = (defaultScreen) => {
  const [screen, setScreen] = useState();

  useEffect( () => {
    if(!defaultScreen) {
      window....
    ...
  return defaultScreen || screen;
}
```

Reply:
Ser ut til å funke fint slik jeg satte det opp? Menyen kollapser i hvert fall fint i portalen når det er implementert slik. Så kan man jo argumentere om det er et fornuftig API eller ikke selvfølgelig.